src/util/session-id.js

```diff
@@ -0,0 +1,121 @@
+/**
+ *
+ *    Copyright (c) 2025 Silicon Labs
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+import { v4 as uuidv4 } from 'uuid'
+import { parse as querystringParse } from 'querystring'
+
+// Session handling constants
+const SESSION_KEY = 'session_uuid'
+const APP_ID_KEY = 'stsApplicationId'
+const DEFAULT_APP_ID = 'defaultAppId'
+
+/**
+ * Loads a session map from the session storage.
+ * Session map is a string=>string key/value map,
+ * which contains "appId" as a key, and the generated UUID as a value.
+ * If appId is not present (as in standalone), then a default value
+ * is used.
+ *
+ * @param {*} window
+ * @returns
+ */
+function loadSessionMap(window) {
+  let json = window.sessionStorage.getItem(SESSION_KEY)
+  if (json == null) {
+    // Nothing there, let's just create an empty map.
+    return new Map()
+  } else {
+    // Found it, so we deserialize it.
+    const parsedArray = JSON.parse(json)
+    return new Map(parsedArray)
+  }
+}
+
+/**
+ * Saves a session map into the session storage.
+ * @param {*} window
+ * @param {*} map
+ */
+function saveSessionMap(window, map) {
+  const mapArray = Array.from(map)
+  const json = JSON.stringify(mapArray)
```

Comment:
Why convert this to json?

Reply:
Because the "sessionStorage" only knows how to save "string => string" object types.
It is not designed to save javascript "Map". So you have to serialize/deserialize it.

Reply:
I stared at it for a while in confusion, until I realized this to make it work......